File: src/test/java/nextstep/ladder/domain/RandomPointGeneratorTest.java
package nextstep.ladder.domain;

import nextstep.ladder.fake.FakeRandom;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.util.Random;

import static org.assertj.core.api.Assertions.assertThat;

class RandomPointGeneratorTest {

    private MovementGenerator createBridge(Random random) {
        return new RandomMovementGenerator(random);
    }

    @Test
    @DisplayName("true 반환 테스트")
    void returnAllTrueTest() {
        MovementGenerator bridgeGenerator = this.createBridge(new FakeRandom(true));
        assertThat(bridgeGenerator.isMove()).isTrue();
    }

    @Test
    @DisplayName("false 반환 테스트")
    void returnAllFalseTest() {
        MovementGenerator bridgeGenerator = this.createBridge(new FakeRandom(false));
        assertThat(bridgeGenerator.isMove()).isFalse();
    }
}

File: src/test/kotlin/ladder/domain/generators/RandomPointGeneratorTest.kt
package ladder.domain.generators

import ladder.domain.Level
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class RandomPointGeneratorTest {
    @Test
    fun `포인트 자동생성` () {
        val points = RandomPointGenerator.generator(5, Level.MEDIUM)
        assertThat(points).hasSize(5)
    }
}
